src/tests/trainer.test.js

```diff
@@ -141,3 +151,74 @@ describe('POST /api/trainers', () => {
     expect(response.error).toBeTruthy();
   });
 });
+
+describe('PUT /api/trainers/:id', () => {
+  test('should return status 400 if the ID is invalid', async () => {
+    const mockId = '@a@invalid@id';
+    const response = await request(app).put(`/api/trainers/${mockId}`).send();
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.message).toMatch(/(invalid)/gi);
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('should return status 404 if is not found', async () => {
+    const mockId = '646428fc0b6aa64a90624c00';
+    const response = await request(app).put(`/api/trainers/${mockId}`).send();
+    expect(response.status).toBe(404);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('should return status 400 if there is other trainer with the same data', async () => {
+    const { _id } = await Trainer.collection.findOne({
+      $or:
+      [
+        { dni: { $ne: mockTrainerWithSameData.dni } },
```

Comment:
You can use directly {$trainerSeed[0].dni} and {$trainerSeed[0].email}. So if the seed changes you don't need to change the mockTrainerWithSameData to test